### PR TITLE
Add support for browser_specific_settings (as noted in docs)

### PR DESCRIPTION
**Pull request by andyrichardson**
## About
The [official documentation](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings) solely states that an addon can specify its `id` in the following manner.

```json
"browser_specific_settings": {
  "gecko": {
    "id": "addon@example.com",
    "strict_min_version": "42.0"
  }
}
```

This ID attribute is not being considered by web-ext. This PR adds support for this feature.

### CODE REVIEW

File: src/util/manifest.js [hunk 1]
```diff
@@ -85,6 +86,20 @@ export default async function getValidatedManifest(
 
 
 export function getManifestId(manifestData: ExtensionManifest): string | void {
-  return manifestData.applications ?
-    manifestData.applications.gecko.id : undefined;
+  if (
+    manifestData.applications &&
+    manifestData.applications.gecko.id
```
Comment (rpl) [inline src/util/manifest.js]: it seems reasonable to also check that the `applications.gecko` property actually exists here, as you are doing for the `browser_specific_settings.gecko` one below.

**Comment by rpl** [comment]
As a side note, this was also failing on npm audit for reasons unrelated to this PR (I already merged #1787 to update the dependency that was triggerring the npm audit failure, and so you should also rebase this PR on top of a recent master to avoid that failure on this pull request)

File: src/util/manifest.js (hunk 1 above)
Comment (andyrichardson) [reply src/util/manifest.js]: I initially added this but then realised the existing code didn't have this check [look like for this reason](https://github.com/mozilla/web-ext/blob/995ccf5b8e651cdfec336be257e908d9c2ee4a03/src/util/manifest.js#L72)

**Comment by andyrichardson** [comment]
Thanks for the feedback. Sounds like this is something that could be merged so I will add some tests around it :+1:

File: src/util/manifest.js (hunk 1 above)
Comment (andyrichardson) [reply src/util/manifest.js]: Updated the PR + types to cover this use case

**Comment by andyrichardson** [comment]
@rpl Will run these through the linter when I get back to my laptop 👍

**Comment by Rob--W** [comment]
@andyrichardson Do you need help to finish this PR?

File: tests/unit/test-util/test.manifest.js [hunk 2]
```diff
@@ -205,29 +215,55 @@ describe('util/manifest', () => {
...
+    const id = 'basic-manifest@web-ext-test-suite';
+
+    it('returns application gecko ID (application.gecko.id)', () => {
+      assert.equal(getManifestId(basicManifest), id);
+    });
 
-    it('returns a gecko ID', () => {
-      assert.equal(getManifestId(basicManifest),
-                   'basic-manifest@web-ext-test-suite');
+    it('returns browser specific gecko ID (browser_specific_settings.gecko.id)', () => {
```
Comment (andyrichardson) [inline tests/unit/test-util/test.manifest.js]:
```suggestion
    it('returns browser_specific_settings.gecko.id', () => {
```

**Comment by rpl** [comment]
Closing in favor of #1944

**Pull request closed without merging**